src/LiveComponent/src/EventListener/DataModelPropsSubscriber.php

```diff
@@ -76,4 +78,19 @@ public static function getSubscribedEvents(): array
             PreMountEvent::class => 'onPreMount',
         ];
     }
+
+    private function getCurrentLiveComponent(ComponentStack $componentStack): ?MountedComponent
+    {
+        $iterator = $componentStack->getIterator();
+        while (($current = $iterator->current()) && !$this->isLiveComponent($current->getComponent()::class)) {
```

Comment:
Because it now implements `IteratorAggregate`, can we just `foreach ($componentStack as $component)`?

Also, can you explain this a little? It's possible that the current component is some *other*, non-live component? And so we need to look further "up"?

Reply:
That's right. If you have for example a, I don't know, Card component as a parent of an Input component, all inside of a Product component, and the input is supposed to control a property of the Product. In HTML this will already be ok, since the product-HTML will have the data-live properties, and the card won't. So, here in PHP, we also need to link it to that Live Component.

I adjusted the comment that was already there.